galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/RestoreCPS.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Licensed Materials - Property of IBM
+ * 
+ * (c) Copyright IBM Corp. 2020.
+ */
+
+package dev.galasa.framework;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.annotations.Component;
+
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IConfigurationPropertyStoreService;
+import dev.galasa.framework.spi.IFramework;
+
+@Component(service = { RestoreCPS.class })
+public class RestoreCPS {
+    
+    private Log             logger  =  LogFactory.getLog(this.getClass());
+    
+    private IFramework      framework;
+    
+    private Path            path;
+    
+    private Map<String, IConfigurationPropertyStoreService>     namespaceCPS 
+                            = new HashMap<String ,IConfigurationPropertyStoreService>();
+    
+    /**
+     * <p>Restores configuration properties from the specified file to the Configuration Property Store (CPS)</p>
+     * 
+     * @param bootstrapProperties
+     * @param overrideProperties
+     * @param filePath
+     * @return
+     * @throws FrameworkException
+     */
+    public void restore(Properties bootstrapProperties, Properties overrideProperties, String filePath) throws FrameworkException {
+        logger.info("Initialising CPS Restore Service");
+        
+        FrameworkInitialisation frameworkInitialisation = null;
+        try {
+            frameworkInitialisation = new FrameworkInitialisation(bootstrapProperties, overrideProperties);
+        } catch (Exception e) {
+            throw new FrameworkException("Unable to initialise the Framework Service", e);
+        }
+        
+        framework = frameworkInitialisation.getFramework();
+        
+        Properties properties = getProperties(filePath);
+        if (properties != null) {
+            restoreProperties(properties);
+        } else {
+            logger.info("No properties found to restore.");
```

Comment:
Why do we want this to trigger a fail in the upstream job? 
Though a real-world example may come across a little contrived (because I can't spot immediate usefulness): It is possible that an empty properties file is intentionally passed to the restore method. In this case, an error might not be appropriate, as the restore may complete correctly without actually restoring any properties.